### PR TITLE
Fix webview theme styles

### PR DESCRIPTION
**Pull request by olexii4**
#### What it does

Add material colors into webviews.

### What issues does this PR fix or reference?
https://github.com/theia-ide/theia/issues/4179

#### How to test

Try to use this 'asciidoctor-vscode' extension
https://marketplace.visualstudio.com/items?itemName=joaompinto.asciidoctor-vscode.
Check all themes.

Screenshot from 2019-09-10 21-06-16
Screenshot from 2019-09-10 21-05-52

### CODE REVIEW

File: packages/core/src/browser/style/index.css [hunk 1]
```diff
@@ -187,7 +187,6 @@ textarea {
 @import './menus.css';
 @import './sidepanel.css';
 @import './tabs.css';
-@import './materialcolors.css';
```
Comment (akosyakov) [inline packages/core/src/browser/style/index.css]: @olexii4 Why it cannot be applied always anymore, but should be applied as part of `useable`?

**Comment by azatsarynnyy** [comment]
I've tested it with K8s extension.

Before:
image

After:
image

Looks much better now :slightly_smiling_face:
Thanks! :+1:

**Comment by akosyakov** [comment]
@azatsarynnyy that's cool!

I just cannot grasp what PR does. It uses and unuses exactly the same styles in core and by doing it somehow fix the issue in the plugin system? It sounds very implicit.

**Comment by AndrienkoAleksandr** [comment]
@akosyakov In the plugin system we display plugin webview in the iframe. And we need provide styles to this iframe.

File: packages/core/src/browser/style/index.css (hunk 1 above)
Comment (olexii4) [reply packages/core/src/browser/style/index.css]: @akosyakov  Previously I added options with id='theia-theme'  to the 'style-loader/useable'  loader to have a possibility to reapply css rules into the webview(inside an iframe). Our thems depends on 'materialcolors.css' that is why we need it inside an iframe.

I try to understand the best way how to do it.  I do not want to add one more id in DOM and configure a new loader webpack or copy and add 'materialcolors.css' in each iframe as a separate resource.

I think the best way to have all variables in the separate <style> tag with id='theia-theme' and I do it.

WDYT?

Comment (akosyakov) [reply packages/core/src/browser/style/index.css]: thank you for the explanation!

So if i understand it should be enough to use materialcolors.css once always not only for dark and light themes?

i.e. instead `static readonly materialColors = require('../../src/browser/style/materialcolors.css');` -> require('../../src/browser/style/materialcolors.css').use()` and better at the top of the file to make sure that it is applied before the shell is loaded?

File: packages/core/src/browser/theming.ts [hunk 2]
```diff
@@ -196,6 +196,7 @@ export class BuiltinThemeProvider {
     // Webpack converts these `require` in some Javascript object that wraps the `.css` files
     static readonly darkCss = require('../../src/browser/style/variables-dark.useable.css');
     static readonly lightCss = require('../../src/browser/style/variables-bright.useable.css');
+    static readonly materialColors = require('../../src/browser/style/materialcolors.css');
```
Comment (akosyakov) [inline packages/core/src/browser/theming.ts]: I think it should go here: https://github.com/theia-ide/theia/blob/ak/plugins_reconnect/packages/core/src/browser/frontend-application-module.ts#L17 and be used always

why -> https://github.com/theia-ide/theia/pull/6155/files#r323145830

**Comment by vinokurig** [comment]
Tested against GitHub plugin:
Before:
screenshot-che-che 192 168 39 115 nip io-2019 09 11-12-48-32
After: 
screenshot-localhost-3000-2019 09 11-12-48-16
Vscode:
Screenshot from 2019-09-11 12-47-52
Looks much better :+1: It is far from the vscode view but at least it is readable.

File: packages/core/src/browser/style/index.css (hunk 1 above)
Comment (olexii4) [reply packages/core/src/browser/style/index.css]: Thanks. I am fixed.

**Comment by olexii4** [comment]
I just provide Theia styles and variables into webview iframes(add materialcolors).  
Now it looks like regular styles for Theia 'p-Widget'  class.
In this place, we can restyle webview iframes depends on variables:
https://github.com/theia-ide/theia/pull/6155/files#diff-2cf6add9fc7566d40d46a3533b4d5cf7R26
And we can customize it.

File: packages/core/src/browser/theming.ts (hunk 2 above)
Comment (olexii4) [reply packages/core/src/browser/theming.ts]: Thanks. Done.

Comment (akosyakov) [reply packages/core/src/browser/theming.ts]: @olexii4 please move the import after this line: https://github.com/theia-ide/theia/blob/ak/plugins_reconnect/packages/core/src/browser/frontend-application-module.ts#L17 to make sure that it is loaded exactly with the same timing, thank you!

**Comment by akosyakov** [comment]
> In this place, we can restyle webview iframes depends on variables:
https://github.com/theia-ide/theia/pull/6155/files#diff-2cf6add9fc7566d40d46a3533b4d5cf7R26

@olexii4 probably it should be a comment in the code

**Comment by olexii4** [comment]
@akosyakov  I have done.

**Comment by kittaakos** [comment]
FYI: the build is broken on the `master`: https://travis-ci.org/theia-ide/theia/builds/583626693

```
lerna ERR! /home/travis/build/theia-ide/theia/packages/core/lib/browser/theming.js:43
1586lerna ERR! require('../../src/browser/style/materialcolors.css').use();
1587lerna ERR!                                                       ^
1588lerna ERR! 
1589lerna ERR! TypeError: require(...).use is not a function
1590lerna ERR!     at Object.<anonymous> (/home/travis/build/theia-ide/theia/packages/core/lib/browser/theming.js:43:55)
1591lerna ERR!     at Module._compile (internal/modules/cjs/loader.js:778:30)
1592lerna ERR!     at Module.replacementCompile (/home/travis/build/theia-ide/theia/node_modules/nyc/node_modules/append-transform/index.js:58:13)
```

Is it related?

**Comment by kittaakos** [comment]
Also:

Screen Shot 2019-09-11 at 14 50 42

Screen Shot 2019-09-11 at 14 51 04

**Comment by AlexTugarev** [comment]
@olexii4 could you please take care of the broken master build?

**Comment by olexii4** [comment]
@AlexTugarev  Ok. I will fix

**Comment by olexii4** [comment]
[PR](https://github.com/eclipse-theia/theia/pull/6165)

**Comment by vince-fugnitto** [comment]
> Also:
> 
> Screen Shot 2019-09-11 at 14 50 42
> 
> Screen Shot 2019-09-11 at 14 51 04


Can we enforce that PRs can not be merged if the CI (Travis and AppVeyor) are failing?
- We can enforce at least that Travis must be passing before enabling the merge, and perhaps AppVeyor as well (since it is much more stable than it used to be)

**Pull request merged**